File: C/Grace/Grace.c
#include <stdio.h>
#include <fcntl.h>
/*
* the aim of this program is to write it's own source code into a file named Grace_kid.c
*/
#define GRACE(x)int main(){ x }
#define STR "#include <stdio.h>%c#include <fcntl.h>%c/*%c* the aim of this program is to write it's own source code into a file named Grace_kid.c%c*/%c#define GRACE(x)int main(){ x }%c#define STR %c%s%c%c#define DUP FILE *fptr = fopen(%cGrace_kid.c%c, %cw%c);char *str = STR;fprintf(fptr, str, 10, 10, 10, 10, 10,10,  34, STR, 34, 10,34, 34, 34, 34, 10);%cGRACE( DUP )"
#define DUP FILE *fptr = fopen("Grace_kid.c", "w");char *str = STR;fprintf(fptr, str, 10, 10, 10, 10, 10,10,  34, STR, 34, 10,34, 34, 34, 34, 10);
GRACE( DUP )

File: JS/Grace/Grace.js
/*
  the aim of this program is to write it's own source code into a file named Grace_kid.c
*/

const str = "/*\n  the aim of this program is to write it's own source code into a file named Grace_kid.c\n*/\n\nconst str = {0};\nconst file = require('fs').createWriteStream('Grace_kid.js');\nconst Grace = () => { file.write(str.replace(/\\{0\\}/g, JSON.stringify(str))); }\n\nGrace();\n";
const file = require('fs').createWriteStream('Grace_kid.js');
const Grace = () => { file.write(str.replace(/\{0\}/g, JSON.stringify(str))); }

Grace();
